src/app/components/Statistics/DateRange/DateRange.tsx

```diff
@@ -63,25 +63,47 @@ const CustomInputContainer = styled.div`
     cursor: pointer;
 `
 
-const DateRange = ({ initialStartDate, InitianEndDate, setEndDate, setStartDate, darkMode }) => {
+const DateRange = ({
+    initialStartDate,
+    InitianEndDate,
+    setEndDate,
+    setStartDate,
+    darkMode,
+    setSeeTimeAxis,
+    disableFuture,
+}) => {
     const { isWideScreenDown, isWidescreen } = useWidth()
 
     const handleClickOneDay = () => {
+        setSeeTimeAxis('day')
         setStartDate(new Date(moment().startOf('day').format('YYYY-MM-DD HH:mm:ss')))
         setEndDate(new Date(moment().endOf('day').format('YYYY-MM-DD HH:mm:ss')))
     }
 
     const handleClickOneMonth = () => {
+        setSeeTimeAxis('month')
         setStartDate(new Date(moment().startOf('month').format('YYYY-MM-DD HH:mm:ss')))
-        setEndDate(new Date(moment().endOf('month').format('YYYY-MM-DD HH:mm:ss')))
+
+        if (!disableFuture) {
+            setEndDate(new Date(moment().endOf('month').format('YYYY-MM-DD HH:mm:ss')))
+        } else {
+            setEndDate(new Date(moment().endOf('day').format('YYYY-MM-DD HH:mm:ss')))
+        }
     }
 
     const handleClickOneYear = () => {
+        setSeeTimeAxis('year')
         setStartDate(new Date(moment().startOf('year').format('YYYY-MM-DD HH:mm:ss')))
-        setEndDate(new Date(moment().endOf('year').format('YYYY-MM-DD HH:mm:ss')))
+
+        if (!disableFuture) {
```

Comment:
I would personally swap this condition, a bit confusing when you negate it here

Reply:
The emissions API fails when sending future dates and when sending the date to the parent's useEffect hook, it queries again so if validation is necessary. Maybe we could change it another way